### ISSUE TITLE
Jumping history in chat with Openfire

### ISSUE THREAD

**Issue by wrooot** [iNPUTmice/Conversations#3355]
#### General information

* **Version:** 2.3.12+pcr
* **Device:** Google Pixel XL (1st gen)
* **Android Version:** Android 9.0 Stock
* **Server name:** igniterealtime.org
* **Server software:** Openfire 4.4.0 alpha/nightly
* **Installed server modules:** Stream Managment, MAM
* **Conversations source:** PlayStore

#### Steps to reproduce

1. Login to server and join Open_Chat room

#### Expected result

No jumping of history.

#### Actual result

Attaching a few videos of the issue:
https://photos.app.goo.gl/xuCcoep6iVZKVMP4A
https://photos.app.goo.gl/J28xXmLYBedJ1P5H7
It seems that this started with yesterday's change to Monitoring plugin, which implements MAM: https://github.com/igniterealtime/openfire-monitoring-plugin/commit/fc7c274d3adbe0c3862b80f8ccc878ce4ffc08c1

**Comment by wrooot** [comment]
Btw, plugin has been downgraded to older version on that server now. So jumping is gone for now. Although it still has issues with history in Conversations. It doesn't pull all the recent history, doesn't show most of the recent live messages. But say Xabber, Spark and some other clients don't have such issues. I guess they are more forgiving to how MAM support is advertised.

**Comment by iNPUTmice** [comment]
> guess they are more forgiving to how MAM support is advertised.

I don’t understand that. I also don’t quite understand the change you linked in the original post? What is Conversations sending and what is OpenFire expecting?

**Comment by wrooot** [comment]
I'm not a developer, so i don't have deep knowledge of how these things work exactly. I meant the other way, maybe Openfire sends/advertises something wrong ant that causes Conversations to behave weirdly. I've been told that downgraded plugin's has a known issue with history/MAM, which maybe will be fixed in later version. But as latest snapshot has been removed there is no way to reproduce the jumping issue (the main issue i wanted to report). So, i'm not sure what to do now. Close this issue or leave it open until newer version of plugin is installed again and it is reproducible again? I have attached the link to the patch hoping maybe at a glance one can see what might be causing Conversations chat window jump forth an back again, but i guess it is not that obvious without knowing how this plugin works.
